internal/storage/metricstore/elasticsearch/factory.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package elasticsearch
+
+import (
+	"flag"
+
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/internal/storage/elasticsearch/config"
+	"github.com/jaegertracing/jaeger/internal/storage/v1"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/metricstore"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/metricstore/metricstoremetrics"
+	"github.com/jaegertracing/jaeger/internal/telemetry"
+)
+
+var _ storage.Configurable = (*Factory)(nil)
+
+type Factory struct {
+	config config.Configuration
+	telset telemetry.Settings
+}
+
+// NewFactory creates a new Factory.
+func NewFactory() *Factory {
```

Comment:
The extension only calls `NewFactoryWithConfig`. Let's minimize the number of different initialization function. Can we just have a single `NewFactory(cfg, telset)`?